### PR TITLE
Add more functions for property selections to enable proper handling of property selections via the API

### PR DESCRIPTION
**Pull request by Jan-Panasiam**

### CODE REVIEW

File: docs/API_reference.md [hunk 1]
```diff
@@ -580,6 +581,24 @@ Return a POST request JSON response, if one of the requests fails return the err
 If the **value_id** field is not filled the method will return: `{'error': 'missing_parameter'}`.
 In case the language within the `lang` field of the JSON is invalid the method will return `{'error': 'invalid_language'}`.
 
+#### plenty_api_create_property_selection_name <a name=create-property-selection></a>
+
+Create a new name for a property selection value and input the initial value for that language.
```
Comment (initBasti) [inline docs/API_reference.md:599]: I think you can reduce the description above to:
`Create a new name for an existing property selection value for the given language`

File: docs/API_reference.md [hunk 2]
```diff
@@ -580,6 +581,24 @@ Return a POST request JSON response, if one of the requests fails return the err
...
+
+[*Required parameter*]:
+
+The **json** parameter contains a single JSON object, containing the selection_Id, language and name.
```
Comment (initBasti) [inline docs/API_reference.md:603]: When you look at existing methods like `plenty_api_create_attribute_value_name` you can see that I prefer to pass in values as parameters if the number of values is very small. I prefer this over a JSON as building JSONs for 3 values is cumbersome for an application and clutters the code. So, please pass these 3 values in as parameters and build the JSON within the library.

File: plenty_api/api.py [hunk 3]
```diff
@@ -1235,6 +1237,31 @@ def plenty_api_create_booking(self, order_id: int,
...
         return response
 
+
+    def plenty_api_create_property_selection_name(self, json: dict) -> dict:
+        """
+        Create a language for a selection property.
+
+        Parameter:
+            json            [dict]  -   single Json object describing the
+                                        creation
```
Comment (initBasti) [inline plenty_api/api.py:1260]: As mentioned above please replace this JSON with the 3 values directly as parameters.

File: plenty_api/api.py [hunk 4]
```diff
@@ -1235,6 +1237,31 @@ def plenty_api_create_booking(self, order_id: int,
                                              data=data)
         return response
 
+
+    def plenty_api_create_property_selection_name(self, json: dict) -> dict:
+        """
+        Create a language for a selection property.
```
Comment (initBasti) [inline plenty_api/api.py:1256]: Wow you create a language :O .. jokes aside, s/language/name in a specific language/

File: plenty_api/api.py [hunk 5]
```diff
@@ -1235,6 +1237,31 @@ def plenty_api_create_booking(self, order_id: int,
...
+        Return:
+                            [dict]  -   Response object if the request should
+                                        fail, the entry contains the error
+                                        message.
+        """
+        if not utils.sanity_check_json(route_name='property_selection',
+                                       json=json):
+            return {'error': 'invalid_json'}
+
+        path = f"/selections/names"
```
Comment (initBasti) [inline plenty_api/api.py:1271]: This doesn't need to be an f"" string, please remove the 'f'.

File: plenty_api/api.py [hunk 6]
```diff
@@ -1235,6 +1237,31 @@ def plenty_api_create_booking(self, order_id: int,
...
+        """
+        Create a language for a selection property.
+
+        Parameter:
+            json            [dict]  -   single Json object describing the
+                                        creation
+
+        Return:
+                            [dict]  -   Response object if the request should
+                                        fail, the entry contains the error
```
Comment (initBasti) [inline plenty_api/api.py:1264]: This sounds like you return the response object only if the request fails.
How about:
```
Return the response object, in case the request fails return an error message
```

File: plenty_api/constants.py [hunk 7]
```diff
@@ -221,12 +223,15 @@
 ]
 REQUIRED_ATTRIBUTE_FIELDS = [('backendName', JSON_STRING)]
 REQUIRED_ATTRIBUTE_VALUE_FIELDS = [('backendName', JSON_STRING)]
+REQUIRED_PROPERTY_SELCTION_FIELDS = [
```
Comment (initBasti) [inline plenty_api/constants.py:226]: s/SELCTION/SELECTION/

**Review by initBasti — CHANGES_REQUESTED** [review]

File: docs/API_reference.md [hunk 8]
```diff
@@ -879,6 +881,28 @@ By specifying a storage location ID within **location_id**, you can target a spe
...
 
+#### Items <a name='put-item-section'></a>
+
+#### plenty_api_update_property_selection_name <a name=update-property-selection></a>
+
+Update a property selection name via its name ID.
+
+[*Required parameter*]:
+
+The **nameId** from Plentymarkets need to be given in order to identify the property selection name that will be updated. The **json** parameter contains a single JSON object, describing the new name.
```
Comment (initBasti) [inline docs/API_reference.md:903]: s/need to be/needs to be/

File: docs/API_reference.md [hunk 9]
```diff
@@ -879,6 +881,28 @@ By specifying a storage location ID within **location_id**, you can target a spe
...
+Please refer to the [Plentymarkets Dev documentation: REST API PUT transaction](https://developers.plentymarkets.com/en-gb/plentymarkets-rest-api/index.html#/).
+
+[*Example*]:
+```json
+{
+    'name': 'value_1',
+}
```
Comment (initBasti) [inline docs/API_reference.md:910]: Please build the JSON within the method and take the name as a parameter.

File: docs/API_reference.md [hunk 10]
```diff
@@ -892,3 +916,4 @@ plenty.cli_progress_bar = True
 variations = plenty.plenty_api_get_variations()
 plenty.cli_progress_bar = False
 ```
+
```
Comment (initBasti) [inline docs/API_reference.md:930]: What happened here? Seems to be unrelated to your changes?

File: plenty_api/api.py [hunk 11]
```diff
@@ -1426,3 +1428,14 @@ def plenty_api_book_outgoing_items(self,
         logging.debug(response)
 
         return response
+
+    def plenty_api_update_property_selection_name(self, name_id: int, json: dict) -> dict:
```
Comment (initBasti) [inline plenty_api/api.py:1443]: Please add a small function description, it doesn't need to be long, but there be a small description.

File: plenty_api/api.py [hunk 12]
```diff
@@ -886,6 +888,15 @@ def plenty_api_get_property_selections(self, refine: dict = None):
 
         return selection_map
 
+    def plenty_api_get_property_selection_names(self, selection_id: int):
+        """
+        Get all Selection IDs and there names for specific selection Id.
```
Comment (initBasti) [inline plenty_api/api.py:893]: Hmm... Get all selection IDs for a specific selection ID?
+ s/there names/their names/
Also, please add the parameters and return part to the description to align with the other methods.

File: docs/API_reference.md [hunk 13]
```diff
@@ -297,6 +298,16 @@ The 'json' format returns a dictionary of property IDs to its available selectio
 example: ```{1: {11: {'de': 'value_1', 'en': 'value_2'}}, 2: {12: {'de': 'value_3', 'en': 'value_4'}}}```
 The 'dataframe' format transforms that data structure into a pandas DataFrame.
 
+##### plenty_api_get_property_selections_names: <a name='get-property-selections-names'></a>
+
+Get all names of a specific selection_id, with additional data such as nameId needed for updating selection names.
```
Comment (initBasti) [inline docs/API_reference.md:303]: s/as nameId/as the nameId/

**Review by initBasti — CHANGES_REQUESTED** [review]

File: docs/API_reference.md (hunk 2 above)
Comment (Jan-Panasiam) [reply docs/API_reference.md:603]: We can do that sure

File: plenty_api/api.py [hunk 14]
```diff
@@ -1235,6 +1237,39 @@ def plenty_api_create_booking(self, order_id: int,
...
+            return [{'error': 'missing_parameter'}]
+
+        if utils.get_language(lang=lang) == 'INVALID_LANGUAGE':
+            return {'error': 'invalid_language'}
+
+        json = {'propertyId': property_id, 'lang': lang, 'name': name,}
+
+        if not utils.sanity_check_json(route_name='property_selection',
+                                       json=json):
+            return {'error': 'invalid_json'}
```
Comment (initBasti) [inline plenty_api/api.py:1286]: I think this sanity check isn't required, I mean you built the JSON so you can trust it as well.

File: plenty_api/constants.py [hunk 15]
```diff
@@ -221,12 +223,15 @@
...
 
 REQUIRED_FIELDS_MAP = {
     'items': REQUIRED_ITEM_FIELDS,
     'variations': REQUIRED_VARIATION_FIELDS,
     'categories': REQUIRED_CATEGORY_FIELDS,
     'attributes': REQUIRED_ATTRIBUTE_FIELDS,
     'attribute_values': REQUIRED_ATTRIBUTE_VALUE_FIELDS,
-    'transaction': REQUIRED_TRANSACTION_FIELDS
+    'transaction': REQUIRED_TRANSACTION_FIELDS,
+    'property_selection': REQUIRED_PROPERTY_SELECTION_FIELDS
```
Comment (initBasti) [inline plenty_api/constants.py:236]: As mentioned above you can delete this sanity check.

File: docs/API_reference.md [hunk 16]
```diff
@@ -580,6 +581,20 @@ Return a POST request JSON response, if one of the requests fails return the err
 If the **value_id** field is not filled the method will return: `{'error': 'missing_parameter'}`.
 In case the language within the `lang` field of the JSON is invalid the method will return `{'error': 'invalid_language'}`.
 
+#### plenty_api_create_property_selection_name <a name=create-property-selection></a>
+
+Create a new name for an existing property selection value for the given language.
+
+[*Required parameter*]:
+
+The **property_id** parameter contains the property Id which a selection name will be created for, the **lang** parameter contains a two letter abbreviation of the target language (for a list of valid values look here: [Language codes](https://developers.plentymarkets.com/en-gb/developers/main/rest-api-guides/getting-started.html#_language_codes)), and the **name** parameter contains the visible name for the selection name in the given language.
```
Comment (initBasti) [inline docs/API_reference.md:603]: s/contains the property Id which a selection name will be created for/
  contains the Plentymarkets ID of a property for which a new selection name value is being added/

**Review by initBasti — CHANGES_REQUESTED** [review]
Unneeded sanity check and a small text improvement.

**Review by initBasti — APPROVED** [review]
